File: src/main/java/com/cfgglobal/test/service/rule/access/CreatorAccessRule.java
package com.cfgglobal.test.service.rule.access;


import com.cfgglobal.test.base.Filter;
import com.cfgglobal.test.domain.Permission;
import com.cfgglobal.test.domain.User;
import org.springframework.stereotype.Component;

@Deprecated
@Component
public class CreatorAccessRule extends AbstractAccessRule {
    @Override
    public Filter exec(Permission permission) {
        User user = securityFilter.currentUser();
        Filter orgFilter = new Filter();
        orgFilter.addCondition("creator.id", user.getId(), Filter.OPERATOR_EQ);
        return orgFilter;
    }

    @Override
    public String getRuleName() {
        return "creator";
    }
}

File: src/main/kotlin/com/cfgglobal/test/service/rule/access/CreatorAccessRule.kt
package com.cfgglobal.test.service.rule.access


import com.cfgglobal.test.domain.Permission
import com.cfgglobal.test.vo.Filter
import org.springframework.stereotype.Component

@Deprecated("")
@Component
class CreatorAccessRule : AbstractAccessRule() {

    override val ruleName: String
        get() = "creator"

    override fun exec(permission: Permission): Filter {
        val user = securityFilter!!.currentUser()
        val orgFilter = Filter()
        orgFilter.addCondition("creator.id", user.id, Filter.OPERATOR_EQ)
        return orgFilter
    }
}
